rust/parquet/src/record/api.rs

```diff
@@ -50,6 +50,33 @@ impl Row {
     pub fn len(&self) -> usize {
         self.fields.len()
     }
+
+    pub fn get_column_iter(&self) -> RowColumnIter {
+        RowColumnIter {
+            fields: &self.fields,
+            curr: 0,
+            count: self.fields.len(),
+        }
+    }
+}
+
+pub struct RowColumnIter<'a> {
+    fields: &'a Vec<(String, Field)>,
+    curr: usize,
+    count: usize,
+}
+
+impl<'a> Iterator for RowColumnIter<'a> {
+    type Item = (usize, &'a String);
```

Comment:
Why do we need the index included in the iterator results? The user could use `enumerate` to get those? Also, would it make more sense for the iterator be over the `Field` rather than just field name?

Reply:
Thanks for the suggestions, I have changed it to return column name and field as a tuple.